app/views/check_csv_file.scala.html

```diff
@@ -14,52 +14,73 @@
  * limitations under the License.
  *@
 
-@import utils.ERSUtil
+@import _root_.utils.ERSUtil
 @import config.ApplicationConfig
-@import uk.gov.hmrc.play.views.html.helpers.FormWithCSRF
+@import views.html.govuk_wrapper
+@import views.html.templates.reference
 
-@this(form: FormWithCSRF)
+@this(
+        govukErrorSummary: GovukErrorSummary,
+        govuk_wrapper: govuk_wrapper,
+        govukBackLink: GovukBackLink,
+        formWithCSRF: FormWithCSRF,
+        govukCheckboxes: GovukCheckboxes,
+        govukButton: GovukButton
+)
 
 @(requestObject: RequestObject, csvFilesList: CsvFilesList)(implicit request: Request[AnyRef], flash: Flash, messages: Messages, ersUtil: ERSUtil, appConfig: ApplicationConfig)
-@import views.html.templates._
+
 @schemeId = @{requestObject.getSchemeId}
 
-@ers_main(title = messages("ers.choose_csv_files.page_title")) {
+@govuk_wrapper(title = messages("ers.choose_csv_files.page_title")) {
     @reference(requestObject.getPageTitle)
-    <p class="font-xsmall"><a href="@routes.CheckFileTypeController.checkFileTypePage" class="link-back">@Html(messages("ers.back"))</a></p>
 
-    @if(flash.get("csv-file-not-selected-error").isDefined){
-        <div class="flash error-summary error-summary--show " id="error-summary-display" role="alert" aria-labelledby="error-summary-heading" tabindex="-1">
-            <h2 id="error-summary-heading" class="h3-heading">@Html(messages("validation.summary.heading"))</h2>
-            <ul class="js-error-summary-messages">
-                <li>
-                    <a href="#error-summary" data-focuses="error-summary">@flash.get("csv-file-not-selected-error")</a>
-                </li>
-            </ul>
-        </div>
-    }
+    @govukBackLink(BackLink(
+        href = routes.CheckFileTypeController.checkFileTypePage.url,
+    ))
+
+    @formWithCSRF(action = routes.CheckCsvFilesController.checkCsvFilesPageSelected()) {
+
+        @if(flash.get("csv-file-not-selected-error").isDefined) {
```

Comment:
just to check with you: does this flash ever kick in post-upscan update?

Reply:
This page isn't the upload page, it's where you select which csv file you want to upload too!